### PR TITLE
【CI】make some test run with old executor in specified windows server

### PR DESCRIPTION
**Pull request by betterpig**
### PR types

Others
### PR changes

Others
### Describe

Some tests will timeout in some windows server, since this PR #42149, it may be caused by new executor, so set these tests use old executor temporarily according to environment variable, which is set "on" in the servers having problem.
details:

- failed tests: 
test_apply_pass_to_program (Timeout) 
test_buffer_shared_memory_reuse_pass (Timeout) 
test_buffer_shared_memory_reuse_pass_and_fuse_optimization_op_pass (Timeout) 
test_imperative_optimizer (Timeout) 
test_imperative_star_gan_with_gradient_penalty (Timeout) 
test_switch_autotune (Timeout) 
test_imperative_mnist_sorted_gradient (Timeout) 
test_custom_relu_op_setup (Timeout) 
test_custom_relu_model (Timeout）
- failed servers:
win6, win7, win10, win12, win14
- problem: not all servers will fail in all failed tests, for example win7 only timeout in test_custom_relu_op_setup, 
test_imperative_star_gan_with_gradient_penalty. What's more weird, if set Glog_v=10, the tests can run successfully, if set Glog_v=0, if hang again.

### CODE REVIEW

**Review by From00 — APPROVED** [review]
LGTM.
The problem is triggered only in a few windows-machines, with strong randomness and weak reproducibility. At present, the several machines that this  issue may occur are offline. In order not to affect the resource adequacy of CI,  we temporarily disable the standalone executor for these UTs on the windows-machines, and will recover them after locating and fixing the problem.

File: python/paddle/fluid/tests/custom_op/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,11 @@ if(WITH_GPU OR APPLE)
     set_tests_properties(test_custom_relu_op_jit PROPERTIES TIMEOUT 180)
     set_tests_properties(test_custom_relu_model PROPERTIES TIMEOUT 180)
     set_tests_properties(test_context_pool PROPERTIES TIMEOUT 180)
+    if($ENV{USE_STANDALONE_EXECUTOR})
+        # these test will fail in some server due to PR#42149, temporarily set it use old executor.
+        set_tests_properties(test_custom_relu_op_setup PROPERTIES ENVIRONMENT FLAGS_USE_STANDALONE_EXECUTOR=0)
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/custom_op/CMakeLists.txt:15]: @From00 这个有上线吗？目前发版默认是`FLAGS_USE_STANDALONE_EXECUTOR=0`吗，这个bug目前影响到用户发版吗

Comment (From00) [reply python/paddle/fluid/tests/custom_op/CMakeLists.txt:15]: 这个bug只在特定几台（看起来性能较差）的windows机器上出现，大概率和机器环境相关，发版默认FLAGS_USE_STANDALONE_EXECUTOR=1，这个已经发了。
这个bug随机性较强，改动log即无法复现，短时间内无法定位出具体问题，需要延期修复。但是这几台机器一直是下线状态，为了不影响windows CI，所以先对这几台机器设置FLAGS_USE_STANDALONE_EXECUTOR=0，后边定位出问题，修复完成后再恢复配置。
触发此问题的PR# https://github.com/PaddlePaddle/Paddle/pull/42149 已经合入2.3，这个PR既是性能优化，也是bug fix，revert反而会引起比较大的影响，所以暂时只能先这样处理。

Comment (zhwesky2010) [reply python/paddle/fluid/tests/custom_op/CMakeLists.txt:15]: @From00 这个应该是底层执行器比较大的问题了，对用户影响比较大，可以暂时先这样跑CI，但下个小版本前最好还是需要修复。@DDDivano 也关注下这个主框架执行器bug

Comment (From00) [reply python/paddle/fluid/tests/custom_op/CMakeLists.txt:15]: @zhouwei25 相关问题已建卡，会高优排查，尽快修复。

**Pull request merged**